test/unit/krane/renderer_test.rb

```diff
@@ -69,10 +69,48 @@ def test_nesting_fields
         c: c3
         d: d4
         foo: bar
-      EOY
-    actual = YAML.dump(YAML.safe_load(render('nest-as-rhs.yaml.erb')))
+    EOY
+    actual = YAML.dump(YAML.safe_load(render("nest-as-rhs.yaml.erb")))
+    assert_equal(expected, actual)
+    actual = YAML.dump(YAML.safe_load(render("nest-indented.yaml.erb")))
+    assert_equal(expected, actual)
+  end
+
+  def test_deployment_id
+    expected = <<~EOY
+      ---
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: migrate-12345678-aaaa
+      spec:
+        containers:
+        - name: migrate
+          image: gcr.io/foobar/api
+    EOY
+    actual = YAML.dump(YAML.safe_load(render("deployment_id.yml.erb")))
     assert_equal(expected, actual)
-    actual = YAML.dump(YAML.safe_load(render('nest-indented.yaml.erb')))
+  end
+
+  def test_renderer_without_current_sha_still_has_a_deployment_id
+    @renderer = Krane::Renderer.new(
+      current_sha: nil,
+      template_dir: fixture_path("for_unit_tests"),
+      logger: logger,
+      bindings: { "a" => "1", "b" => "2" },
+    )
+    expected = <<~EOY
+      ---
+      apiVersion: v1
+      kind: Pod
+      metadata:
+        name: migrate-aaaa
```

Comment:
Without  a task_id or a a current_sha set you're going to get random bits afte the `-`. Did this pass locally for you?

Reply:
Oh oops I messed up and missed committing the mocha call I was making locally. Was having some trouble getting my editor to agree with the committed code format and was selectively staging, force pushed the correct code.